lib/cloud/azure/vm.go

```diff
@@ -94,7 +94,10 @@ func (c *vmClient) Get(ctx context.Context, resourceID string) (*VirtualMachine,
 
 	var identities []Identity
 	if resp.Identity != nil {
-		identities = append(identities, Identity{ResourceID: *resp.Identity.PrincipalID})
+		if systemAssigned := StringVal(resp.Identity.PrincipalID); systemAssigned != "" {
+			identities = append(identities, Identity{ResourceID: systemAssigned})
```

Comment:
@gabrielcorado 
Do we need to add a system assigned identity to `Identities` pool that are used as SQLServer database users ? 
Maybe we can create a `VirtualMachine.SystemIdenitty` field where only a user assigned identify will be considered as valid db users ?

Reply:
I've kept it because this is a more generic client; if someone needs this information, they can still use it.

Reply:
note that `Identity.PrincipalID` is not a valid `ResourceID` and will fail to parse with `arm.ParseResourceID` - this prevents SQL Server login using system assigned identity from ever working.

Principal ID looks like: `11111111-2222-3333-4444-555555555555`

User assigned Identity keys from the response are valid resource ID like: 
```
/subscriptions/11111111-2222-3333-4444-555555555555/resourceGroups/database-access/providers/Microsoft.ManagedIdentity/userAssignedIdentities/sqlserver-alice
````

From https://learn.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview the name of a system-assigned managed identity is the same as the resource it was created for. However, I tested creating a managed identity with the same name as my Azure VM and it allowed me to do so and to assign it as a user-assigned managed identity. So there is potentially ambiguity if a system-assigned identity and user-assigned managed identity have the same name. I'd rather we just advise customers that using the system-assigned managed identity as a login is not supported.

Furthermore, we should handle the case where there is no default managed identity better and include info about that in the troubleshooting docs. See info about default identity here: https://learn.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/managed-identities-faq#what-identity-will-imds-default-to-if-dont-specify-the-identity-in-the-request relevant bit: ` If system assigned managed identity isn’t enabled, and multiple user assigned managed identities exist, then you are required to specify a managed identity in the request.`

I worked around the issue by setting environment variables AZURE_TENANT_ID and AZURE_CLIENT_ID to match my tenant and the managed identity that has discovery permissions, and that resolved the issue. The alternative is to just use a system-assigned managed identity and give that discovery permissions.

Reply:
Currently, it won't work with system-managed identities because we require it to be in the `ResourceID` format. We could make it more explicit in the code by separating user-managed and system-managed, as @smallinsky mentioned. Although, I'm not sure if we can provide a better error message for this case, and the guide we currently have only mentioned that they need to give a user-managed identity. Do you think we should change that?

I've created an issue for that second part where we should have better error handling when the SDK cannot define which identity to use: https://github.com/gravitational/teleport/issues/20513.

Reply:
@gabrielcorado I think we should change the guide to advise them to use a system assigned managed identity for the discovery permissions, and then use user assigned identities for each SQL Server login they want to create.

For code changes: we should change the code for `vmClient.Get` such that it puts all the user assigned identities in one field and the system-assigned identity's prinicipal ID in a a separate field inside the `VirtualMachine` struct.